### ISSUE TITLE
ghc843 is missing

### ISSUE THREAD

**Issue by rubenmoor** [cprussin/nixjs#5]
I am on `nix-channel 19.03` and I use this `shell.nix`

```
{
    purescript ? "0.12.4",
    nixjs-version ? "0.0.10",
    nixjs ? fetchTarball "https://github.com/cprussin/nixjs/archive/${nixjs-version}.tar.gz",
    nixpkgs ? <nixpkgs>
}:

let
    nixjs-overlay = import nixjs { inherit purescript; };
    pkgs = import nixpkgs { overlays = [ nixjs-overlay ]; };
in

pkgs.mkShell {
    buildInputs = [ pkgs.purescript ];
}

```

Executing `nix-shell` fails with `attribute ghc843` is missing. I do have ghc 8.6.4 installed. Is there a way to pick a ghc that I have either installed already or one that is available? What am I doing wrong?

**Comment by cprussin** [comment]
Hm that's interesting, I actually hadn't built using nixjs on 19.03 but I literally just switched over so I'm sure I'll run into this... let me try a build now and see if I can figure this one out.  Thanks for the heads up @rubenmoor !

**Comment by cprussin** [comment]
**tl;dr**: Upgrade to nixjs `0.0.12` and it should work out of the box.

So I remember why I locked the ghc version, there's some issue with compiling purescripts on later versions of ghc.  I haven't quite figured out what the root cause is--something to do with

```
Configuring cabal-doctest-1.0.6...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:968:20 in Cabal-2.5.0.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:466:12 in Cabal-2.5.0.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:597:20 in Cabal-2.5.0.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:67:5 in Cabal-2.5.0.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:179:19 in Cabal-2.5.0.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-2.5.0.0:Distribution.Simple
  defaultMain, called at Setup.hs:2:8 in main:Main
Setup: Encountered missing dependencies:
Cabal >=1.10 && <2.5
```

If I recall correctly, there's some known issue in either one of purescript's dependencies itself, or in the way it's packaged for nix.  I'm honestly not sure and would need to do more research to remember.  In any case, while I'd like a better fix to be able to use `ghcHEAD`, in the meantime I've just switched to `ghc844` which should work on 19.03.

I'd be open to a PR to make the ghc version configurable, but I don't personally have a use case for it and would rather it just work out of the box, so I didn't take that approach with my change.
